libevmasm/Assembly.cpp

```diff
@@ -1682,8 +1691,13 @@ LinkerObject const& Assembly::assembleEOF() const
 	// DATALOADN loads 32 bytes from EOF data section zero padded if reading out of data bounds.
 	// In our case we do not allow DATALOADN with offsets which reads out of data bounds.
 	auto const staticAuxDataSize = maxAuxDataLoadNOffset.has_value() ? (*maxAuxDataLoadNOffset + 32u) : 0u;
-	solRequire(preDeployDataSectionSize + staticAuxDataSize < std::numeric_limits<uint16_t>::max(), AssemblyException,
-		"Invalid DATALOADN offset.");
+
+	if (preDeployDataSectionSize + staticAuxDataSize > std::numeric_limits<uint16_t>::max())
```

Comment:
Before, the case where `preDeployDataSectionSize + staticAuxDataSize == std::numeric_limits<uint16_t>::max()` would trigger an error, but with the change made in the PR, it will not.
Not sure which behavior is right here.

Reply:
Good point. The current version is correct. According to EOF spec the size of the data section must be in range `0x0000-0xFFFF` so `std::numeric_limits<uint16_t>::max()` is still a valid value.

Looks like this was a bug that slipped through in #15456. I played with this check a bit here and I thought I restored it to the previous state but looks like I unintentionally fixed the bug while doing so :)

I extracted the fix into its own PR: #15667.